src/solution.jl

```diff
@@ -1,42 +1,49 @@
-# build OCP solution from DOCP solution
-function OCPSolutionFromDOCP(ipopt_solution, docp)
-
-    # save general solution data
-    docp.NLP_stats = ipopt_solution
-    if is_min(docp.ocp)
-        docp.NLP_objective = ipopt_solution.objective
-    else
-        docp.NLP_objective = - ipopt_solution.objective
-    end
-    docp.NLP_constraints_violation = ipopt_solution.primal_feas
-    docp.NLP_iterations = ipopt_solution.iter
-    docp.NLP_solution = ipopt_solution.solution
-    docp.NLP_sol_constraints = zeros(docp.dim_NLP_constraints)
-    ipopt_constraint!(docp.NLP_sol_constraints, ipopt_solution.solution, docp)
+# build OCP solution from DOCP solution (GenericExecutionStats)
+# ipopt solution is a GenericExecutionStats
+# https://jso.dev/SolverCore.jl/dev/reference/#SolverCore.get_status-Tuple{NLPModels.AbstractNLPModel}
+function OCPSolutionFromDOCP(docp, docp_solution_ipopt)
+
+    # could pass some status info too (get_status ?)
+    return OCPSolutionFromDOCP_raw(docp, docp_solution_ipopt.solution, objective=docp_solution_ipopt.objective, constraints_violation=docp_solution_ipopt.primal_feas, iterations=docp_solution_ipopt.iter,multipliers_constraints=docp_solution_ipopt.multipliers, multipliers_LB=docp_solution_ipopt.multipliers_L, multipliers_UB=docp_solution_ipopt.multipliers_U, message=docp_solution_ipopt.solver_specific[:internal_msg])
+end
+
+# still missing: stopping and success info...
+function OCPSolutionFromDOCP_raw(docp, solution; objective=nothing, constraints_violation=nothing, iterations=0, multipliers_constraints=nothing, multipliers_LB=nothing, multipliers_UB=nothing, message=nothing)
```

Comment:
```suggestion
function OCPSolutionFromDOCP_raw(docp, solution; objective=DOCP_objective(solution, docp), constraints_violation=nothing, iterations=0, multipliers_constraints=nothing, multipliers_LB=nothing, multipliers_UB=nothing, message=nothing)
```

I did not check all optional arguments, but I saw that objective remains `nothing` if it is not properly define.
`DOCP_objective(get_variable(solution, docp), docp)` should do the trick.
Maybe it can be directly integrated in `parse_DOCP_solution()` (but that would require duplication if `objective` is already properly set).

Reply:
Thanks, added as well as the constraints violation. Done in the function body since the constraints part is a bit more involved (need to check both variables and constraints bounds)